android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt

```diff
@@ -211,14 +234,22 @@ class AddonsManagerAdapter(
     }
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal fun bindFooterButton(
-        holder: FooterViewHolder,
-    ) {
+    internal fun bindFooterButton(holder: FooterViewHolder) {
         holder.itemView.setOnClickListener {
             addonsManagerDelegate.onFindMoreAddonsButtonClicked()
         }
     }
 
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun bindHeaderButton(holder: HeaderViewHolder) {
+        holder.restartButton.setOnClickListener {
+            engine.enableExtensionProcessSpawning()
```

Comment:
We may want to move this where we are handling the `ProcessEnabledAction`, maybe we could create a middleware to have all logic changes there.

Reply:
I am not sure to follow why we'd want to move this out.